components/cloud/azure/src/azblob.rs

```diff
@@ -443,9 +441,9 @@ impl ContainerBuilder for TokenCredContainerBuilder {
             {
                 let token_response = self.token_cache.read().unwrap();
                 if let Some(ref t) = *token_response {
-                    let interval = t.0.expires_on - Utc::now();
+                    let interval = (t.0.expires_on - OffsetDateTime::now_utc()).whole_minutes();
```

Comment:
Does it compile with OffsetDateTime? 
Also make sure it works when expires_on < OffsetDateTime::now_utc() by adding a test case.